src/index.ts

```diff
@@ -176,14 +176,17 @@ export function createBirpc<RemoteFunctions = {}, LocalFunctions = {}>(
         await _promise
         return new Promise((resolve, reject) => {
           const id = nanoid()
-          rpcPromiseMap.set(id, { resolve, reject })
-          post(serialize(<Request>{ m: method, a: args, i: id, t: 'q' }))
+          let timeoutId
+
           if (timeout >= 0) {
-            setTimeout(() => {
+            timeoutId = setTimeout(() => {
               reject(new Error(`[birpc] timeout on calling "${method}"`))
               rpcPromiseMap.delete(id)
-            }, timeout)
+            }, timeout).unref?.()
```

Comment:
.unref might need to be called conditionally as birpc will also run on browsers.

Reply:
`setTimeout` should always return something in browsers. Isn't condition for `unref?` enough?

https://developer.mozilla.org/en-US/docs/Web/API/setTimeout#return_value

Reply:
Yeah, true